src/runtime.js

```diff
@@ -151,6 +151,7 @@ let tornado = {
    */
   exists(val, placeholderNode, bodies, context) {
     if (this.util.isPromise(val)) {
+      placeholderNode = this.insertPendingBody(placeholderNode, bodies.pending, context) || placeholderNode;
```

Comment:
what makes `eq` worthy of being a helper?   Why does `?exists` not live as a helper? IMO, having `?` look at the data makes the distinction muddy. 

for more muddiness why not have it as a filter?

`{#reddit.dat  | pending}`